src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/BackgroundService.cs

```diff
@@ -78,6 +89,9 @@ public virtual async Task StopAsync(CancellationToken cancellationToken)
 
         }
 
+        /// <summary>
+        /// Tells the service task to stop but does not wait for it to stop.
```

Comment:
I think this should be documented like other `Dispose` methods. 

```suggestion
        /// Releases all resources used by the <see cref="BackgroundService" /> object.
```

Reply:
But the implementation does what I have written.  Why would you want to hide this information to the user?

Reply:
One reason is because it may change in the future. Writing xml summary comments that say exactly what the implementation does doesn't scale. Instead, the xml summary comments should describe the high-level intent on what a method is for / does. If we wanted to have a method that only "Tells the service task to stop but does not wait for it to stop.", we would name the method `StopWithoutWaiting`, or similar. But here the method is named "Dispose" because it "Releases all resources used by the BackgroundService object".

Reply:
We can add additional detailed information to the `remarks` section, if that information is useful to the user.

Reply:
The member `_executeTask` is [IDisposable](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable) but it will not be disposed of, so your description looks factually inaccurate.

Reply:
If you are interested, check out https://devblogs.microsoft.com/pfxteam/do-i-need-to-dispose-of-tasks/.

Reply:
Why I appreciate the info, it is still wrong to say you release something when you do not.  Even if it is not necessary to release it.

Reply:
The wording is the standard "Dispose" description:

https://docs.microsoft.com/dotnet/api/system.io.stream.dispose
https://docs.microsoft.com/dotnet/api/system.threading.timer.dispose
https://docs.microsoft.com/dotnet/api/system.threading.tasks.task.dispose

Adding our docs experts to weigh in on how to doc Dispose methods: @carlossanlop @gewarren @BillWagner @IEvangelist

Reply:
We have specific wording here: https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Method.

```suggestion
        /// Releases the resources used by the current instance of the <see cref="BackgroundService" /> class.
```

Reply:
We also have the standard way of implementing [Dispose](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable.disposeSystem_IDisposable_Dispose) which the standard wording describes and which this particular implementation does not follow.

Reply:
I agree that this class doesn't implement the Dispose pattern correctly, but it has already shipped and it can't be fixed. Either way, that is a separate topic.

Reply:
What cannot be fixed should at least be documented.

Reply:
I'm all for consistency. I'd rather keep it as suggested from Eric: https://github.com/dotnet/runtime/pull/63011#discussion_r777688269. I believe it is unnecessary to explicitly state every `private` type in the triple slash comments that is disposed of. Simply stating that it releases the resources is appropriate as that's what it does.

Reply:
The method does not release any resources.